docs/tutorial/code/flask/task.yaml

```diff
@@ -0,0 +1,212 @@
+# IMPORTANT
+# Comments matter!
+# The docs use the wrapping comments as
+# markers for including said instructions
+# as snippets in the docs.
+###########################################
+summary: Getting started with Flask tutorial
+
+kill-timeout: 90m
+
+environment:
+
+execute: |
+  #move everything to $HOME so that Juju deployment works
+  mv *.yaml *.py *.txt $HOME
+  cd $HOME
+  #don't use the staging store for this test
+  unset CHARMCRAFT_STORE_API_URL
+  unset CHARMCRAFT_UPLOAD_URL
+  unset CHARMCRAFT_REGISTRY_URL
+  #make sure that Juju version is correct
+  snap install rockcraft --classic
+  snap install lxd
+  lxd init --auto
+  snap refresh charmcraft --channel=latest/edge --amend
+  snap install microk8s --channel=1.31-strict/stable
+  #adduser $USER snap_microk8s
+  #newgrp snap_microk8s
+  snap refresh juju --channel=3.5/stable --amend
+  lxc network set lxdbr0 ipv6.address none
+  mkdir -p ~/.local/share
+  microk8s status --wait-ready
+  microk8s enable hostpath-storage
+  microk8s enable registry
+  microk8s enable ingress
+  juju bootstrap microk8s dev-controller
+
+  # [docs:create-venv]
+  sudo apt-get update && sudo apt-get install python3-venv -y
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r requirements.txt
+  # [docs:create-venv-end]
+
+  flask run -p 8000 &
+  retry -n 5 --wait 2 curl --fail localhost:8000
+
+  # [docs:curl-flask]
+  curl localhost:8000
+  # [docs:curl-flask-end]
+
+  kill $!
+
+  # [docs:create-rockcraft-yaml]
+  rockcraft init --profile flask-framework
+  # [docs:create-rockcraft-yaml-end]
+  sed -i "s/name: .*/name: flask-hello-world/g" rockcraft.yaml
+  sed -i "s/amd64/$(dpkg --print-architecture)/g" rockcraft.yaml
+
+  # [docs:pack]
+  rockcraft pack
+  # [docs:pack-end]
+
+  # [docs:ls-rock]
+  ls *.rock -l
+  # [docs:ls-rock-end]
+
+  # [docs:skopeo-copy]
+  rockcraft.skopeo --insecure-policy copy --dest-tls-verify=false \
+    oci-archive:flask-hello-world_0.1_$(dpkg --print-architecture).rock \
+    docker://localhost:32000/flask-hello-world:0.1
+  # [docs:skopeo-copy-end]
+
+  # [docs:create-charm-dir]
+  mkdir charm
+  cd charm
+  # [docs:create-charm-dir-end]
+
+  # [docs:charm-init]
+  charmcraft init --profile flask-framework --name flask-hello-world
+  # [docs:charm-init-end]
+
+  # [docs:charm-pack]
+  charmcraft pack
+  # [docs:charm-pack-end]
+
+  # [docs:ls-charm]
+  ls *.charm -l
+  # [docs:ls-charm-end]
+
+  # [docs:add-juju-model]
+  juju add-model flask-hello-world
+  # [docs:add-juju-model-end]
+
+  juju set-model-constraints -m flask-hello-world arch=$(dpkg --print-architecture)
+
+  # [docs:deploy-juju-model]
+  juju deploy ./flask-hello-world_ubuntu-22.04-$(dpkg --print-architecture).charm \
+    flask-hello-world --resource \
+    flask-app-image=localhost:32000/flask-hello-world:0.1
+  # [docs:deploy-juju-model-end]
+
+  # [docs:deploy-nginx]
+  juju deploy nginx-ingress-integrator --channel=latest/edge --revision 122
```

Comment:
Any particular reason for pinning the revision please? I think `lates/edge` should get the latest revision

Reply:
`latest/edge` did not get the latest revision in the GitHub CI test -- it did for my local testing though.

In GitHub, `latest/edge` grabbed revision 44 (suspiciously, this is the latest revision for the Ubuntu 22.04 `base` while 122 is the latest revision for the Ubuntu 20.04 `base`) and revision 44 ran into `hook failed: "config-changed"` errors upon deployment

Reply:
I think that is an issue with the charm, I don't know why we are running it on 20.04. Let's leave it for now and I'll see whats going on

Reply:
Is there a preference between specifying the `base` or the `revision` here? I guess maybe it doesn't matter so much since it's a temporary fix in any event?

Reply:
The base would be better so that we keep tracking the latest revision automatically